extensions-core/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsDataSegmentPusher.java

```diff
@@ -107,18 +108,23 @@ public DataSegment push(File inDir, DataSegment segment) throws IOException
     final DataSegment dataSegment;
     try (FSDataOutputStream out = fs.create(tmpFile)) {
       size = CompressionUtils.zip(inDir, out);
-      final Path outFile = new Path(String.format(
+      Path outFile = new Path(String.format(
           "%s/%s/index.zip",
           fullyQualifiedStorageDirectory,
           storageDir
       ));
+      StringBuilder outFileUriStringBuilder= new StringBuilder(outFile.toUri().toString());
+      outFileUriStringBuilder.setCharAt(outFileUriStringBuilder.lastIndexOf("/"), '_');  //replaces last `/` with `_`
```

Comment:
can we instead add methods in DataSegmentPusherUtil so that it gives the segment location upto version and partition number is added here appropriately instead of doing the string replace etc